check_diff/src/lib.rs

```diff
@@ -335,3 +340,25 @@ pub fn compile_rustfmt(
         feature_runner,
     });
 }
+
+pub fn check_diff(config: Option<Vec<String>>, runners: CheckDiffRunners) -> i32 {
+    let mut errors = 0;
+    for entry in WalkDir::new(".").into_iter().filter_map(|e| e.ok()) {
+        let path = entry.path();
+        if path.is_file() && path.extension().map_or(false, |ext| ext == "rs") {
+            match runners.create_diff(path, &config) {
+                Ok(diff) => {
+                    if !diff.is_empty() {
```

Comment:
This is what I was getting at earlier. We need a convenient way to check if the diff is empty. I haven't worked with the `diffy` crate before so I don't know what it outputs when there is no diff. Does it still include the diff header like

```
--- a
+++ b
```

or is it just an empty string?

It would be great if you could investigate that / create a unit test for that behavior.

Reply:
investigated. the result is 
```
--- a
+++ b
```

furthermore, when there is a diff, we get something as such:
```
--- a
+++ b
@@ -1 +1 @@
-The quick brown fox jumps over the lazy dog
+The quick brown fox jumps over the LAZY dog
```

I think the `diffy` crate tells us that `-` means that those lines belongs in file `a` while those that contains `+` belongs in file `b`. I took the chance to have a look at the `similar` crate as well. For `similar`, the output is less verbose (which is perhaps something we want?), for the same output, we get the following:
```
-The quick brown fox jumps over the lazy dog
+The quick brown fox jumps over the LAZY dog
```

Also i realised that the goal of `diffy` actually helps to resolves such conflicts too. I am not sure whether this is a functionality that we want to use. If it is something that we don't require, perhaps it may be better to have a cleaner "diff" by using the `similar` crate instead.